lib/view/draw_badge_screen.dart

```diff
@@ -0,0 +1,162 @@
+import 'package:badgemagic/providers/drawbadge_provider.dart';
+import 'package:badgemagic/view/widgets/navigation_drawer.dart';
+import 'package:badgemagic/virtualbadge/view/draw_badge.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:get_it/get_it.dart';
+import 'package:provider/provider.dart';
+
+class DrawBadge extends StatefulWidget {
+  const DrawBadge({super.key});
+
+  @override
+  State<DrawBadge> createState() => _DrawBadgeState();
+}
+
+class _DrawBadgeState extends State<DrawBadge> {
+  DrawBadgeProvider cellStateToggle = GetIt.instance<DrawBadgeProvider>();
+
+  @override
+  void initState() {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.landscapeLeft,
+    ]);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+      DeviceOrientation.portraitDown,
+      DeviceOrientation.landscapeLeft,
+      DeviceOrientation.landscapeRight,
+    ]);
+    cellStateToggle.resetGrid();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    DrawBadgeProvider drawToggle = Provider.of<DrawBadgeProvider>(context);
+    SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersiveSticky);
+    return Scaffold(
+      appBar: AppBar(
+        leading: Builder(builder: (context) {
+          return IconButton(
+              onPressed: () {
+                Scaffold.of(context).openDrawer();
+              },
+              icon: const Icon(
+                Icons.menu,
+                color: Colors.white,
+              ));
+        }),
+        backgroundColor: Colors.red,
+        title: const Text(
+          'Badge Magic',
+          style: TextStyle(color: Colors.white),
+        ),
+      ),
```

Comment:
You should create a common class which will encapsulate this logic, rather than duplicating it. We need to manage the drawer from one class. You can then have these hooks which inject the content and creates the entire page for you